packages/vite/src/utils/executor-utils.ts

```diff
@@ -45,7 +46,14 @@ export function createBuildableTsConfig(
       context.targetName === 'serve' ? 'build' : context.targetName,
       context.configurationName
     );
-    // this tsconfig is used via the vite ts paths plugin
-    createTmpTsConfig(tsConfig, context.root, projectRoot, dependencies);
+    // This tsconfig is used via the Vite ts paths plugin.
+    // It can be also used by other user-defined Vite plugins (e.g. for creating type declaration files).
+    const tmpTsConfigPath = createTmpTsConfig(
+      tsConfig,
+      context.root,
+      projectRoot,
+      dependencies
+    );
+    process.env.NX_TSCONFIG_PATH = tmpTsConfigPath;
```

Comment:
When `options.buildLibsFromSource === true`, `process.env.NX_TSCONFIG_PATH` would be undefined. Is that desired?

Reply:
it's expected - `process.env.NX_TSCONFIG_PATH` would also be undefined during `test` or `serve` in case of app.
so the consumer would have to always check it like:
```
const tsconfigPath = process.env.NX_TSCONFIG_PATH ?? join(__dirname, 'tsconfig.lib.json')
```

whether it's desired I'm not that certain - it seems to me to be better for consumers than doing something like:
```
const tmpTsconfigPath = join(
  workspaceRoot,
  'tmp',
  projectRootFromWorkspaceRoot,
  'tsconfig.generated.json'
)
const tsconfigPath = existsSync(tmpTsconfigPath) ? tmpTsconfigPath : join(__dirname, 'tsconfig.lib.json')
```

both of these are leaking implementation details, but the first one is a little less leaky imho